ts/Core/Chart/Chart.ts

```diff
@@ -3782,7 +3782,12 @@ class Chart {
             // It is not necessary to calculate extremes on ordinal axis,
             // because they are already calculated, so we don't want to override
             // them.
-            if (!axis.isOrdinal || scale !== 1 || reset) {
+            if (
+                !axis.isOrdinal ||
+                !axis.ordinal?.getExtendedPositions() || // #21316
```

Comment:
The comment above says the ticks should be already calculated - could you double-check this? I would rather avoid bloating Core with code related to Stock

Reply:
Yeah, I agree with this one. How about we just do this?
```suggestion
                axis.options.overscroll || // #21316
```

Tests are passing, I've tested it manually - everything seems to be fine 🤔

Reply:
Looks much better. Green light from me 👍